### PR TITLE
ethclient: add DialContext and Close

### PR DESCRIPTION
**Pull request by asymmetric**
This PR adds:

* `ethclient.DialContext`, which is useful to handle cancellation
* `ethclient.Close`, which is useful to manually close the underlying connection

Both of these are just wrappers around the underlying implementations in the `rpc` package.

### CODE REVIEW

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -51,6 +55,10 @@ func NewClient(c *rpc.Client) *Client {
 	return &Client{c}
 }
 
+func (ec *Client) Close() {
+	ec.Close()
```
Comment (fjl) [inline ethclient/ethclient.go]: This is an infinite loop ;)

**Comment by fjl** [comment]
We could improve the documentation of Dial to make this more clear though.

File: ethclient/ethclient.go (hunk 1 above)
Comment (asymmetric) [reply ethclient/ethclient.go]: You are obviously right, and it's a typo :). Fixed.

**Comment by asymmetric** [comment]
@fjl I think that once you implement this thin wrapper, adding `DialContext` just makes it more useful. if `ethclient.Dial` is for "quick connections in one-off scripts where you don't care much about timeouts", `ethclient.DialContext` is for ""quick connections in one-off scripts where you **do** care about timeouts", which is [my case](https://github.com/asymmetric/beehive/blob/asymmetric/ethereum/bees/ethereumbee/ethereumbee.go#L45): I would like to time-out the underlying connection after n seconds.

**Comment by asymmetric** [comment]
Also considering that other methods in the package [do use](
https://github.com/asymmetric/go-ethereum/tree/fe6cf00f480aa6347b8d55079cbe6c473482ea27/ethclient/ethclient.go#L79) `context.Context`.

**Comment by vancoldgfunk** [comment]
thank for your support but my goal is to fin that fast transaction
connexion not lose any control of that . i try to build that ETHereum to
buy my package on promotion contest , but every step i go true from
knowledge base , my created token is not value yet for real token . but i
got sound and available ETH to create and have real fast connexion
availability. what should i do on the go to have that fast because you tell
me is for fast transfer . i need to ready that immediately if you
understand what i go treu let me re- enter my own and eurning . available
that site today ethereumcashpro.com.

2018-03-16 12:56 GMT-04:00 Lorenzo Manacorda <notifications@github.com>:

**Comment by karalabe** [comment]
Please document new methods next time :)

**Pull request merged**